pkgs/racket-test/tests/racket/contract/contract-out.rkt

```diff
@@ -1958,4 +1958,33 @@
                #`'#,(struct-field-info-list (syntax-local-value #'foo))))
       (eval '(extract-field-names)))
    (list 'x))
+
+  (test/spec-passed/result
+   'provide/contract-upe-struct-transformer
+   '(begin
+      (eval '(module provide/contract-upe-struct-transformer racket/base
+               (require racket/contract/base)
+               (provide (contract-out
+                         #:unprotected-submodule no-contract
+                         (struct s ([x integer?]))))
+               (struct s (x) #:constructor-name make-s)))
+      (eval '(require racket/match
+                      (submod 'provide/contract-upe-struct-transformer no-contract)))
+      (eval '(match (make-s 1) [(s x) x])))
+   1)
+
```

Comment:
It is probably better to put this into another module to avoid having stuff defined at the top-level that could potentially mess with other tests.